Makefile.in

```diff
@@ -0,0 +1,108 @@
+PROJECT := ulwgl
+
+OBJDIR := build
+
+PREFIX ?= /usr
+BINDIR := $(PREFIX)/bin
+LIBDIR := $(PREFIX)/lib
+DATADIR := $(PREFIX)/share
+
+DESTDIR ?=
+USERINSTALL ?= xfalse
+
+
+.PHONY: all
+all: reaper ulwgl ulwgl-launcher
+
+
+$(OBJDIR)/.build-ulwgl: | $(OBJDIR)
+	$(info :: Building ulwgl )
+	sed 's|##INSTALL_PATH##|$(DATADIR)/$(PROJECT)|g' ULWGL/ulwgl-run.in > $(OBJDIR)/ulwgl-run
+	touch $(@)
```

Comment:
The ulwgl-run files have been deleted since 050ce8237306feff08ec69cc32abc0eabe219fd5 because the bug related to script's parent has been fixed.

That being said, compared to simply installing a symbolic link, why are you still choosing to create a shell script instead?

Reply:
I am have a very strong preference towards not using symlinks in `bin/` directories for packaging. I do not have any strong arguments against using them though, probably I have long forgotten the reason I developed this preference.

One that comes to mind is that in a chain of symlinks, like we do here, the entry point gets lost leading to subtle issues like the one https://github.com/Open-Wine-Components/ULWGL-launcher/commit/050ce8237306feff08ec69cc32abc0eabe219fd5 is fixing. One thing that worries me is that the `Makefile` itself is not advanced enough for what the the weirder distros might do, so having an internal entry point could help in that regard. The third one is that if this becomes a `setuptools` project, `ulwgl-run` will become a file anyways.

That being said, since I do not have good reasons, I updated it to whatever we decide.

Reply:
Creating a symbolic at /usr/bin suffices and is proven to be relatively safe as a lot of applications/distributions currently already do this so there should not be any concern for future problems. The reason the entry point (/usr/share/ULWGL) got lost was due to not understanding that Path objects do not automatically resolve symbolic links on property access and due to insufficient testing.

I believe a simple Makefile suffices for this case, but I just wondered if Meson can make the process simpler or be a better fit... 

However, what weird distributions specifically do you have in mind that worries you that makes creating a shell script a better fit? I'm not following how automating the creation of a symbolic link in the Makefile to the correct path relates to issues for distributors nor how symbolic links are too advanced for Makefiles.

Reply:
> as a lot of applications/distributions currently already do this so there should not be any

I did look into this and most symlinks in bin seem to be targeting files in the same directory `/usr/bin/something` -> `/usr/bin/something_else`

> However, what weird distributions specifically do you have in mind that worries you that makes creating a shell script a better fit?

NixOS, but I guess we can cross that bridge when we come to it. Since I do not know enough about it and we shall find out once it they try to package it, so I'll convert it into a symlink.

>  I'm not following how automating the creation of a symbolic link in the Makefile to the correct path relates to issues for distributors nor how symbolic links are too advanced for Makefiles

Maybe I am overthinking this, let's go with the symlink but keep the templates in the repo in case we need them.

Reply:
There is one problem with symlinks though, if someone uses `install` to copy the file into their whatever correct location, the source is going to be copied instead of the link itself.

Reply:
I converted it into a symlink. The issue that myself encountered doing this is that if someone tries to use the unix `install` program to copy the file into their location of choice, the source file instead of the link will be copied.

Reply:
Sorry, but where are you doing this and when is there a case when a user will need to do this often?

I believe users invoking install or forgetting the -P option for the `cp` command is not our concern as we're defining a strict Makefile to use for them. With it, we're saving them the work by automating the building and file organization of this application so distributors should be solely be relying it for building it. Therefore, it will be on them if the source file instead of the link gets copied since they are deciding to do their own thing by not using the Makefile.

Reply:
The default behavior of `cp` is simply problematic enough for me to not consider symlinking for this case any more. The makefile goes up to the point of installing the files in a specified directory for the building software to find them and compress them, we do not control what happens between `make install` and the package.  Since we do not know how this will be packaged, I prefer to be safe than sorry. It might be on them, but I do not want to create a pitfall that I know exists.

Reply:
Sorry, I believe this is out of our concern that needlessly complicates things.

 cc @GloriousEggroll on whether this is a valid and practical concern for packagers

Reply:
Sorry, but this issue has become a bit of bike-shedding without any real value. This doesn't affect how anything else runs and it is safer for the reasons I described above and more "proper". There is no good argument against it either. Can we please let this go since there is no reason to change it?